admin/services/data-access/ps-report-cache.data.service.js

```diff
@@ -1,11 +1,23 @@
 'use strict'
-
 const PsReportCache = require('../../models/ps-report-cache')
 const psReportCacheDataService = {}
 
-psReportCacheDataService.create = async function (data) {
-  const model = new PsReportCache(data)
-  return model.save()
+/**
+ * Save a document, replacing any existing document
+ * @param data
+ * @return {Promise.<*>} returns a mongo response doc `{ n: 1, nModified: 1, ok: 1 }`
+ */
+psReportCacheDataService.save = async function (doc) {
+  return PsReportCache.update({_id: doc._id}, doc, {upsert: true}).exec()
```

Comment:
if you remove upsert wouldn't that work regardless?

Reply:
No.  The save() method can be called on a document without an `_id` - and save it.  It can also be called with a document with an `_id` in which case it updates the document.

Upset = update or insert